lib/AssemblyUtil/FastaToAssembly.py

```diff
@@ -0,0 +1,258 @@
+        return assembly_data
+
+
+
+    def filter_contigs_by_length(self, fasta_file_path, min_contig_length):
+        ''' removes all contigs less than the min_contig_length provided '''
+        filtered_fasta_file_path = fasta_file_path + '.filtered.fa'
+
+        fasta_record_iter = SeqIO.parse(fasta_file_path, 'fasta')
+        short_seq_iterator = (record for record in fasta_record_iter if len(record.seq) >= min_contig_length)
+        SeqIO.write(short_seq_iterator, filtered_fasta_file_path, 'fasta')
+
+        return filtered_fasta_file_path
+
+
+    def save_assembly_object(self, workspace_id, assembly_name, obj_data):
+        print('Saving Assembly to Workspace')
+        sys.stdout.flush()
+        obj_info = self.dfu.save_objects({'id': workspace_id,
+                                          'objects': [{'type': 'KBaseGenomeAnnotations.Assembly',
+                                                       'data': obj_data,
+                                                       'name': assembly_name
+                                                       }]
+                                          })[0]
+        return obj_info
+
+
+    def save_fasta_file_to_shock(self, fasta_file_path):
+        ''' Given the path to the file, upload to shock and return Handle information
+            returns:
+                typedef structure {
+                    string shock_id;
+                    Handle handle;
+                    string node_file_name;
+                    string size;
+                } FileToShockOutput;
+
+        '''
+        print('Uploading fasta file (' + str(fasta_file_path) + ') to SHOCK')
+        sys.stdout.flush()
+        return self.dfu.file_to_shock({'file_path': fasta_file_path, 'make_handle': 1})
+
+
+    def stage_input(self, params):
+        ''' Setup the input_directory by fetching the files and returning the path to the file'''
+        file_path = None
+        if 'file' in params:
+            file_path = os.path.abspath(params['file']['path'])
+        if 'shock_id' in params:
+            print('Downloading file from SHOCK node: ' + str(params['shock_id']))
+            sys.stdout.flush()
+            input_directory = os.path.join(self.scratch, 'assembly-upload-staging-' + str(uuid.uuid4()))
+            os.makedirs(input_directory)
+            file_name = self.dfu.shock_to_file({'file_path': input_directory,
+                                                'shock_id': params['shock_id']
+                                                })['node_file_name']
+            file_path = os.path.join(input_directory, file_name)
+        if 'ftp_url' in params:
+            print('Downloading file from: ' + str(params['ftp_url']))
+            sys.stdout.flush()
+            file_path = self.dfu.download_web_file({'file_url': params['ftp_url'],
+                                                    'download_type': 'FTP'
+                                                    })['copy_file_path']
```

Comment:
It'll raise an exception at the end of the function if 'file_path' is not set, so I don't think we need this.